File: dev-docs/snippets/documents/is-synced.ts
import type { JsBaoClient } from "js-bao-wss-client";

// Check whether a document's local state is synced with the server (local,
// synchronous).
export function isSynced(client: JsBaoClient, documentId: string) {
  // #region example
  const synced = client.documents.isSynced(documentId);
  // #endregion example
  return synced;
}

File: dev-docs/snippets/documents/is-synced.swift
import JsBaoClient

// Check whether a document's local state is synced with the server — a cheap
// synchronous local read. (inSync(...) is the async server round-trip.)
func isSynced(client: JsBaoClient, documentId: String) {
  // #region example
  let synced = client.documents.isSynced(documentId: documentId)
  // #endregion example
  _ = synced
}
